Dockerfile.s3

```diff
@@ -48,7 +48,9 @@ ARG TARGETARCH
 RUN --mount=type=cache,target=/var/cache/apt \
     apt-get update && \
         # Install dependencies to add additional repos
-        apt-get install -y gcc
+        apt-get install -y gcc && \
+        # Workaround to fix JPype1 compile error on ARM Linux "gcc: fatal error: cannot execute ‘cc1plus’"
+        apt-get install -y g++
```

Comment:
I wished this was not necessary, especially not needed for S3 😿

Reply:
I guess this would be a bit faster if it would be in a single `apt-get install` call? 🤔

Reply:
And maybe this should be added somewhere in the contribution guide? Because it seems Mac users will struggle to run `make install` without `g++` (and Java when they use this feature) on their machines?

Reply:
+1 for a single apt-get call, updated ✅ 

The `g++` dependency might be specific to ARM Linux because it worked out of the box for me on an ARM MacBook.

I agree that the Java installation and setting `JAVA_HOME` is a bit cumbersome in host mode 😿
@alexrashed Where is the right place to document service-specific requirements for host mode? (we have Notion, the dev guide on the website, getting started guide)

Reply:
I think this should be part of the contribution guidelines (because it affects all open source contributors), which is currently located in our docs.

Reply:
Why do we need it for S3? Because we put it into `base-runtime`? Could we just move the dependency into `runtime` instead? I'm not sure I entirely follow here.

Reply:
We don't need it for S3. Thanks for suggesting the switch to `runtime` deps 👍 
Fixed as commented in https://github.com/localstack/localstack/pull/10615#discussion_r1570279927